src/MonacoEditor.tsx

```diff
@@ -1,18 +1,41 @@
-import React, { useRef, useEffect } from "react";
+import React, { useRef, useEffect, useState } from "react";
 import * as monaco from "monaco-editor/esm/vs/editor/editor.api";
 import { KeyCode, KeyMod } from "monaco-editor";
 import { format } from "prettier/standalone";
 import tsParser from "prettier/parser-typescript";
+import * as ts from "typescript";
+import { hashString } from "./utils";
+
+// TODO: use this to implement undo (CMD + Z) in the editor
+const UNDO_TRACKER = 4;
+const lscache = {
+  __store: new Map(),
```

Comment:
This can also be a `Set` for better semantics but it doesn't make a difference in this case.